### PR TITLE
NGDC AE (211) Instrument

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Addresses the memory issue in #88 by adapting the load routine for the AE data to only load the desired portion of the 10-year data set.

# Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality or documentation)
- This change requires a documentation update

# How Has This Been Tested?

```
import datetime as dt
import pysat
import pysatMadrigal as pymad

ae = pysat.Instrument(inst_module=pymad.instruments.ngdc_ae, user='Your Name', password='your.email@inst.place')
ae.download(start=pymad.instruments.ngdc_ae.madrigal_start)
ae.load(date=dt.datetime(1981, 1, 1))

print(ae.variables)
```

Yields: 
```
Index(['year', 'month', 'day', 'hour', 'min', 'sec', 'recno', 'kindat',
       'kinst', 'ut1_unix', 'ut2_unix', 'aem', 'alm', 'aum', 'aom', 'smpint',
       'md', 'hm', 'hmi', 'ae', 'al', 'au', 'ao'],
      dtype='object')
```

## Test Configuration
* Operating system: OS X Big Sur
* Version number: Python 3.9
* Any details about your local setup that are relevant: develop branch of pysat

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have linted the files updated in this pull request
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Comment by jklenzing** [comment]
It looks like I forgot to set the the `pip_rc_install` workflow to `workflow_dispatch` before merging back in #86. Recommend fixing this at some point. Not necessarily now since the other jobs take so long to run.  Review pending completion of tests.

File: pysatMadrigal/instruments/madrigal_pandas.py [hunk 1]
```diff
@@ -100,15 +100,12 @@
...
-# function
 tag_dates = {'120': dt.datetime(1963, 11, 27), '170': dt.datetime(1998, 7, 1),
-             '210': dt.datetime(1950, 1, 1), '211': dt.datetime(1978, 1, 1),
-             '212': dt.datetime(1957, 1, 1), '7800': dt.datetime(2009, 11, 10)}
+             '210': dt.datetime(1950, 1, 1), '212': dt.datetime(1957, 1, 1),
+             '7800': dt.datetime(2009, 11, 10)}
 _test_dates = {'': {tag: tag_dates[tag] if tag in tag_dates.keys()
                     else tag_dates['7800'] for tag in tags.keys()}}
-_test_download = {'': {tag: True for tag in tags.keys()}}
+_test_download = {'': {tag: tag in tag_dates.keys() for tag in tags.keys()}}
```
Comment (aburrell) [inline pysatMadrigal/instruments/madrigal_pandas.py:108]:
```suggestion
_test_download = {'': {tag: True if tag in tag_dates.keys() else False for tag in tags.keys()}}
```

**Comment by aburrell** [comment]
@jklenzing can you create an issue or PR for that?

**Comment by jklenzing** [comment]
> @jklenzing can you create an issue or PR for that? You could also just make the change in #88.

Added as #90. Will probably add it to the PR shortly

**Comment by jklenzing** [comment]
This branch: 319 passed, 1 skipped, 67 warnings in 6837.05s (1:53:57)
develop: 316 passed, 1 skipped, 67 warnings in 17442.61s (4:50:42)

**Review by jklenzing — APPROVED** [review]
Code is working as expected. Added #91 as a possible future route to extend this to other applicable instruments.

**Pull request merged**